src/comm/ConnOpener.cc

```diff
@@ -34,7 +34,7 @@ Comm::ConnOpener::ConnOpener(const Comm::ConnectionPointer &c, const AsyncCall::
     AsyncJob("Comm::ConnOpener"),
     host_(NULL),
     temporaryFd_(-1),
-    conn_(c),
+    conn_(c->cloneDestinationDetails()),
```

Comment:
Don't we need cloneIdentDetails() for the specific "Ident" case?

Reply:
We do! Moreover, we want to copy nfmark, nfConnmark, and tos because most if not all ConnOpener creators set some (or all) of those, right?

I suspect we should replace cloneDestinationDetails() and cloneIdentDetails() with a single cloneProfile() call that will do what cloneIdentDetails() does now and use cloneProfile() instead of the old two methods everywhere. What do you think?

Reply:
>  most if not all ConnOpener creators set

Agreed, for example Ftp::Client::connectDataChannel() and HappyConnOpener::openFreshConnection().

> I suspect we should replace cloneDestinationDetails() and cloneIdentDetails() with a single cloneProfile() call

We probably can copy tos and nfmark, but I am a little worried about nfConnmark flag, which has some limitations for FTP:
```
 In case of FTP, the CONNMARK will NOT be applied to data connections [...]
```

Reply:
ConnOpener clones the Connection object just to break the possible link with the caller. The caller is still responsible for correctly configuring the Connection it gives to ConnOpener. If FTP code needs to do something special, it should do it. Afterwords, ConnOpener should use (a copy of) whatever Connection object it was given (rather than decide which parts of that object to copy into the clone). That is why we need cloneProfile().

>  In case of FTP, the CONNMARK will NOT be applied to data connections

Yes, but that is done by never calling Ip::Qos::setNfConnmark() for the FTP data connection and possibly (I have not checked) by not even setting nfConnmark on the FTP data connection. In other words, that special case does exist, but it is not related to the communication between ConnOpener and its creator, including the FTP code that creates ConnOpener for the data connection.

I am working on fixing this bug.

Reply:
Should be fixed now. Primary commit: 5b1d7d0.